Source/MQTTnet/Server/MqttServerKeepAliveMonitor.cs

```diff
@@ -0,0 +1,119 @@
+﻿using MQTTnet.Diagnostics;
+using MQTTnet.Internal;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MQTTnet.Implementations;
+using MQTTnet.Protocol;
+
+namespace MQTTnet.Server
+{
+    public sealed class MqttServerKeepAliveMonitor
+    {
+        readonly IMqttServerOptions _options;
+        readonly MqttClientSessionsManager _sessionsManager;
+        readonly IMqttNetScopedLogger _logger;
+
+        public MqttServerKeepAliveMonitor(IMqttServerOptions options, MqttClientSessionsManager sessionsManager, IMqttNetLogger logger)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _sessionsManager = sessionsManager ?? throw new ArgumentNullException(nameof(sessionsManager));
+            
+            if (logger == null) throw new ArgumentNullException(nameof(logger));
+            _logger = logger.CreateScopedLogger(nameof(MqttServerKeepAliveMonitor));
+        }
+
+        public void Start(CancellationToken cancellationToken)
+        {
+            // The keep alive monitor spawns a real new thread (LongRunning) because it does not 
+            // support async/await. Async etc. is avoided here because the thread will usually check
+            // the connections every few milliseconds and thus the context changes (due to async) are 
+            // only consuming resources. Also there is just 1 thread for the entire server which is fine at all!
+            Task.Factory.StartNew(_ => DoWork(cancellationToken), cancellationToken, TaskCreationOptions.LongRunning).Forget(_logger);
+        }
+
+        void DoWork(CancellationToken cancellationToken)
+        {
+            try
+            {
+                _logger.Info("Starting keep alive monitor.");
+
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    TryMaintainConnections();
+                    PlatformAbstractionLayer.Sleep(_options.KeepAliveMonitorInterval);
```

Comment:
i would use task.delay and pass in the cancellationtoken. why use blocking sleep

Reply:
My intention is that the thread is started as "long running" and thus is not relying in available threads in the thread pool when using async/await.